swerve100/src/main/java/frc/robot/commands/DriveWithHeading.java

```diff
@@ -88,11 +88,10 @@ public void execute() {
             snapMode = true;
             desiredRotation = pov.getRadians();
         }
-
         currentPose = m_robotDrive.getPose();
         double currentRads = MathUtil.angleModulus(currentPose.getRotation().getRadians());
 
-        if (snapMode && Math.abs(rotSwitch) < 0.1) {
+        if (snapMode && Math.abs(rotSwitch) < 0.1 && AHRSClass.gyrosWorking) {
```

Comment:
statics like this are bad, instead pass a reference to the gyro in the constructor, assign it to a member variable, and use that variable here.